Assets/Scripts/GrappleGunBehavior.cs

```diff
@@ -125,8 +150,11 @@ private bool GrappleBroken()
     private Vector3 ComputeTugForce()
     {
         var grappleDirection = GrapplePoint() - player.transform.position;
-        var tugStrength = grappleDirection.magnitude / maxGrappleLength * maxRetractionForce;
-        var tugForce = grappleDirection.normalized * tugStrength;
+        var distFromGrapplePoint = grappleDirection.magnitude;
+        
+        var directionMultiplier = Mathf.Abs(Vector3.Dot(playerRb.velocity.normalized, grappleDirection.normalized) - 1);
+        var tensionMultiplier = 35f * directionMultiplier * Mathf.Clamp(distFromGrapplePoint / maxRetractionForce, 0.5f, 1);
```

Comment:
Can you walk me through the math for the `directionMultiplier` and `tensionMultiplier` and what the goal of each is / how its achieved?

Reply:
...pls?

Reply:
Basically, it applies the same boost the thruster does when the grapple is pulling you in an opposite direction of your current velocity. What it translates to is that it whips you backwards harder (more rigid rope).